src/raftstore/store/worker/pd.rs

```diff
@@ -176,6 +184,50 @@ impl<T: PdClient> Runner<T> {
             error!("report split failed {:?}", e);
         }
     }
+
+    fn handle_get_region(&self, local_region: metapb::Region, peer: metapb::Peer) {
+        metric_incr!("pd.get_region");
+        match self.pd_client.get_region(local_region.get_start_key()) {
```

Comment:
Currently PD uses (startkey, endkey) ranges as the keys to index the region infos in memory.  Give it region id to query the corresponding region info would require it to scan all meta data.
Calling `get_region` by `start_key` is a possible way to do this PR without changing PD dramatically.